src/main/webapp/app/plagiarism/manage/exercise-update-plagiarism/exercise-update-plagiarism.component.ts

```diff
@@ -1,89 +1,99 @@
+        this.form = this.fb.group({
+            continuousPlagiarismControlEnabled: [false],
+            continuousPlagiarismControlPostDueDateChecksEnabled: [false],
+            similarityThreshold: [null, [Validators.required, Validators.min(0), Validators.max(100)]],
+            minimumScore: [null, [Validators.required, Validators.min(0), Validators.max(100)]],
+            minimumSize: [null, [Validators.required, Validators.min(0), Validators.max(100)]],
+            continuousPlagiarismControlPlagiarismCaseStudentResponsePeriod: [null, [Validators.required, Validators.min(7), Validators.max(31)]],
+        });
 
-    ngAfterViewInit(): void {
-        this.fieldCPCEnabledSubscription = this.fieldCPCEnabled?.valueChanges?.subscribe(() => this.calculateFormValid());
-        this.fieldTreshholdSubscription = this.fieldThreshhold?.valueChanges?.subscribe(() => this.calculateFormValid());
-        this.fieldMinScoreSubscription = this.fieldMinScore?.valueChanges?.subscribe(() => this.calculateFormValid());
-        this.fieldMinSizeSubscription = this.fieldMinSize?.valueChanges?.subscribe(() => this.calculateFormValid());
-        this.fieldResponsePeriodSubscription = this.fieldResponsePeriod?.valueChanges?.subscribe(() => this.calculateFormValid());
-    }
+        this.formStatus = toSignal(this.form.statusChanges, { initialValue: this.form.status });
 
-    ngOnDestroy() {
-        this.fieldCPCEnabledSubscription?.unsubscribe();
-        this.fieldTreshholdSubscription?.unsubscribe();
-        this.fieldMinScoreSubscription?.unsubscribe();
-        this.fieldMinSizeSubscription?.unsubscribe();
-        this.fieldResponsePeriodSubscription?.unsubscribe();
-    }
+        this.isFormValid = computed(() => this.formStatus() === 'VALID');
+
+        this.form.get('continuousPlagiarismControlEnabled')!.valueChanges.subscribe((enabled: boolean) => {
+            const fields = [
+                'continuousPlagiarismControlPostDueDateChecksEnabled',
+                'similarityThreshold',
+                'minimumScore',
+                'minimumSize',
+                'continuousPlagiarismControlPlagiarismCaseStudentResponsePeriod',
+            ];
+            fields.forEach((key) => {
+                const ctrl = this.form.get(key);
+                if (enabled) {
+                    ctrl?.enable();
+                } else {
+                    ctrl?.disable();
+                }
+            });
+        });
 
-    calculateFormValid(): void {
-        this.formValid = Boolean(
-            !this.exercise.plagiarismDetectionConfig?.continuousPlagiarismControlEnabled ||
-                (this.fieldThreshhold?.valid && this.fieldMinScore?.valid && this.fieldMinSize?.valid && this.fieldResponsePeriod?.valid),
-        );
-        this.formValidChanges.next(this.formValid);
+        effect(() => {
+            this.formValid = this.isFormValid();
+            this.formValidChanges.next(this.formValid);
```

Comment:
Why do we need a Subject `formValidChanges ` when we have a `formValid` signal ?

I do think that the occurrences subscribing to `formValidChanges` should be exchanged with an effect that is triggered on `formValid` changes instead - this is the point of the signals migrations, isn't it?